presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/SfmSketchMergeAggregation.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.operator.aggregation.noisyaggregation.sketch.SfmSketch;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.AggregationFunction;
+import com.facebook.presto.spi.function.AggregationState;
+import com.facebook.presto.spi.function.CombineFunction;
+import com.facebook.presto.spi.function.InputFunction;
+import com.facebook.presto.spi.function.OutputFunction;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.type.SfmSketchType;
+import io.airlift.slice.Slice;
+
+import static com.facebook.presto.common.type.VarbinaryType.VARBINARY;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+@AggregationFunction("merge")
+public final class SfmSketchMergeAggregation
+{
+    private SfmSketchMergeAggregation() {}
+
+    @InputFunction
+    public static void input(@AggregationState SfmSketchState state, @SqlType(SfmSketchType.NAME) Slice value)
+    {
+        SfmSketch sketch = SfmSketch.deserialize(value);
+        SfmSketch previous = state.getSketch();
+        if (previous == null) {
+            // if state sketch is empty, add this sketch to the state
+            state.setSketch(sketch);
+            state.setEpsilon(0); // not used
+            state.addMemoryUsage(sketch.getRetainedSizeInBytes());
+        }
+        else {
+            // if state already has a sketch, merge in the current sketch
+            try {
+                // throws if the sketches are incompatible (e.g., different bucket count/size)
+                previous.mergeWith(sketch);
```

Comment:
Correct, it doesn't change the footprint. I don't anticipate it changing in the future, but I assume it's cheap enough to make the calls that we can do it anyway just in case.

Reply:
> I don't anticipate it changing in the future...

Oops, I changed it already! Now that I've updated `Bitmap` to wrap Java's `BitSet`, the actual length of serialized bytes is variable-length. In practice, this means that serialized state will vary in size and should be a good bit smaller during much of the typical aggregation, since the bits in the bitmap will concentrate in the lower positions. The final result will usually be about the same size, though, since random bits will be scattered throughout the bitmap.